template/apps/api/src/resources/webhook/actions/stripe-webhook.ts

```diff
@@ -0,0 +1,70 @@
+import config from 'config';
+import Stripe from 'stripe';
+import { AppKoaContext, AppRouter, Next } from 'types';
+
+import { stripeService } from 'services';
+import { userService, userHelper } from 'resources/user';
+import { subscriptionHelper } from 'resources/subscription';
+
+interface ValidatedData {
+  event: Stripe.Event;
+}
+
+async function validator(ctx: AppKoaContext<ValidatedData>, next: Next) {
+  const signature = ctx.request.header['stripe-signature'];
+
+  ctx.assertClientError(signature, { signature: 'Stripe signature header is missing' }, 403);
+
+  try {
+    const event = stripeService.webhooks.constructEvent(ctx.request.rawBody, signature, config.stripe.webhookSecret);
+
+    ctx.validatedData.event = event;
+  } catch (err: any) {
+    ctx.status = 400;
+    ctx.message = `Webhook Error: ${err.message}`;
+    return null;
+  }
+
+  await next();
+}
+
+async function handler(ctx: AppKoaContext<ValidatedData>) {
+  const { event } = ctx.validatedData;
+
+  switch (event.type) {
+    case 'setup_intent.succeeded':
+      const setupIntent = event.data.object as Stripe.SetupIntent;
+      await Promise.all([
+        userHelper.updateDefaultPaymentMethod({
+          customer: setupIntent.customer as string,
+          paymentMethod: setupIntent.payment_method as string,
+        }),
+        subscriptionHelper.updateSubscriptionPaymentMethod({
+          customer: setupIntent.customer as string,
+          paymentMethod: setupIntent.payment_method as string,
+        }),
```

Comment:
Let's remove these helpers and create handler for 'setup_intent.succeeded' inside separate file near `stripe-webhook.ts`